mycroft/skills/skill_updater.py

```diff
@@ -45,228 +101,85 @@ class SkillUpdater:
                                 with the mycroft core system and handle
                                 commands.
     """
-    _installed_skills_file_path = None
-    _msm = None
 
     def __init__(self, bus=None):
+        self.selene_skills = SeleneDeviceSkills()
```

Comment:
Should `selene_skills` init be optional? Or perhaps the name refactored since Selene/similar backends are optional

Reply:
this specifically follows the Selene spec to upload the metadata, its wraps the msm functionality that was removed, this is  selene specific to get things in its expected data format

the actually uploading should be optional like the whole backend is. Alternative backends would not work either way if the url is set to them unless they expect this same data format

Reply:
Why init the `SeleneDeviceSkills` module at all if there's no Selene-like backend available?

Reply:
it is still used and called by code, its methods are expected, just wrapped into a class for cleanliness instead of being all over the main object and unused

this does not handle the upload, only the expected api for generating the data etc. I could name it something else if the name is causing confusion. This is the manifest file for skills you have installed created on device. theoretically it could be consumed by other 3d party apps

Reply:
it used to require msm installed to generate this

i wrapped it into a json database so we have direct control over the final file, but this simply represents a file that mycroft needs to generate on device

Reply:
Okay, maybe it is just the name; would it make sense to call this `SkillsManifest` or something to indicate it isn't directly tied to Selene?

Reply:
I didn't know this was here. Why not use it for osm?

Reply:
this is for backwards compat, i would burn it with fire if i could, not use it in osm :P 

this does not contain useful info for OSM, it is very selene specific data format and data points with garbage like device identifiers in the mix and not respecting skill_id and so on

Reply:
im actually forcing skill ids in some places where a "nice name" was generated so make the data cleaner, this is where msm was used before, to provide a "nice name"

Reply:
in fact, i will make this private, dont want to introduce this in the public api at all!